src/engine/strat_engine/engine.rs

```diff
@@ -85,23 +87,14 @@ impl StratEngine {
         let mut table = Table::default();
         let mut incomplete_pools = HashMap::new();
 
-        for (pool_uuid, devices) in &pools {
-            match StratPool::setup(&dm, *pool_uuid, devices) {
+        for (pool_uuid, devices) in pools {
+            match setup_pool(&dm, pool_uuid, &devices, &table) {
                 Ok((pool_name, pool)) => {
-                    let evicted = table.insert(pool_name, *pool_uuid, pool);
-                    if evicted.is_some() {
-                        // TODO: update state machine on failure.
-                        let _ = teardown_pools(&dm, table);
-                        let err_msg = "found two pools with the same name";
-                        return Err(EngineError::Engine(ErrorEnum::Invalid, err_msg.into()));
-                    }
+                    table.insert(pool_name, pool_uuid, pool);
```

Comment:
We are not checking to see if anything is evicted with this change.  I think we can convince ourselves that this isn't possible based on the code flow in getting here (that we don't have a conflicting UUID), but should we assert that?

Reply:
Can assert here if desired...should assert in ```block_evaluate()``` as well if that is the plan.